core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedElementNameUtil.java

```diff
@@ -20,53 +20,37 @@
 
 import java.lang.reflect.Field;
 import java.lang.reflect.Parameter;
+import java.util.function.Function;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Ascii;
 import com.google.common.base.CaseFormat;
 
+import com.linecorp.armeria.server.annotation.Attribute;
 import com.linecorp.armeria.server.annotation.Header;
 import com.linecorp.armeria.server.annotation.Param;
 
 final class AnnotatedElementNameUtil {
 
     /**
-     * Returns the value of the {@link Param} annotation which is specified on the {@code element} if
-     * the value is not blank. If the value is blank, it returns the name of the specified
-     * {@code nameRetrievalTarget} object which is an instance of {@link Parameter} or {@link Field}.
+     * Returns the value of {@link Header}, {@link Param}, {@link Attribute} if the value is not blank.
+     * If the value is blank, it returns the name of the specified {@code nameRetrievalTarget} object
+     * which is an instance of {@link Header}, {@link Param}, {@link Attribute} or {@link Field}.
      */
-    static String findName(Param param, Object nameRetrievalTarget) {
+    static String findName(Object nameRetrievalTarget, String value, Function<String, String> delegate) {
         requireNonNull(nameRetrievalTarget, "nameRetrievalTarget");
-
-        final String value = param.value();
         if (DefaultValues.isSpecified(value)) {
             checkArgument(!value.isEmpty(), "value is empty.");
             return value;
         }
-        return getName(nameRetrievalTarget);
+        return delegate.apply(getName(nameRetrievalTarget));
     }
 
-    /**
-     * Returns the value of the {@link Header} annotation which is specified on the {@code element} if
-     * the value is not blank. If the value is blank, it returns the name of the specified
-     * {@code nameRetrievalTarget} object which is an instance of {@link Parameter} or {@link Field}.
-     *
-     * <p>Note that the name of the specified {@code nameRetrievalTarget} will be converted as
-     * {@link CaseFormat#LOWER_HYPHEN} that the string elements are separated with one hyphen({@code -})
-     * character. The value of the {@link Header} annotation will not be converted because it is clearly
-     * specified by a user.
-     */
-    static String findName(Header header, Object nameRetrievalTarget) {
-        requireNonNull(nameRetrievalTarget, "nameRetrievalTarget");
-
-        final String value = header.value();
-        if (DefaultValues.isSpecified(value)) {
-            checkArgument(!value.isEmpty(), "value is empty.");
-            return value;
-        }
-        return toHeaderName(getName(nameRetrievalTarget));
+    static String findName(Object nameRetrievalTarget, String value) {
```

Comment:
We could move this method before `static String findName(Object nameRetrievalTarget, String value, Function<String, String> delegate)` because its method signature is simpler while having the same name.

Reply:
Thanks your comment 🙇‍♂️ 
I see! i moved it!